### PR TITLE
feat(kbreadcrumb): add icon slot

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

Add icon slot for each breadcrumb (needed for Runtime Manager breadcrumbs).

image

## PR Checklist

* [x] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [x] **Tests pass:** check the output of yarn test
* [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [x] **Framework style:** abides by the essential rules in Vue's style guide
* [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/components/breadcrumbs.md [hunk 1]
```diff
@@ -148,6 +148,50 @@ Content to be displayed between breadcrumb items, defaults to a chevron.
 </KBreadcrumbs>
 ```
 
+### breadcrumb-icon
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply docs/components/breadcrumbs.md:151]: It already is that except you switched whether we lead with `icon` or end with it? I am following the approach we used for the item icon slot in `KTreelist`. I don't understand what you're asking for.

[the remark this answers is not in the document]
Comment (kaiarrowood) [reply docs/components/breadcrumbs.md:151]: I can switch `icon` to lead, but like I said it already is using just `icon-{key}`. The only time `breadcrumb` is thrown in is when there is no key and we are using the index, which respectfully I think `icon-breadcrumb-1` is more intuitive than `icon-1`, so I will keep.

[the remark this answers is not in the document]
Comment (kaiarrowood) [reply docs/components/breadcrumbs.md]: It isn't POLA because it is explained in the docs. We haven't ever made `key` required previously and you aren't required to use the icon slot so it doesn't make sense to make `key` required now.

File: docs/components/breadcrumbs.md [hunk 2]
```diff
@@ -231,6 +289,11 @@ export default defineComponent({
     font-size: 13px;
     font-weight: 300;
     line-height: 14px;
-    color: var(--kui-color-text-neutral-weak, $kui-color-text-neutral-weak);
+    color: #afb7c5;
```
Comment (DariaYeremina) [inline docs/components/breadcrumbs.md:292]: question: why did you remove a token?

Comment (kaiarrowood) [reply docs/components/breadcrumbs.md:292]: @portikM 's guide said not to use tokens in the docs

**Comment by DariaYeremina** [comment]
LGTM ⭐

File: docs/components/breadcrumbs.md (hunk 2 above)
Comment (portikM) [reply docs/components/breadcrumbs.md:292]: * correction, we're definitely not using `var()` in docs. If you want - you can use the SCSS variable like so:
```
color: $kui-color-text-neutral-weak;
```

But the sure way is to just use hardcoded value - perfectly fine in docs.

**Review by DariaYeremina — APPROVED** [review]
